cheroot/connections.py

```diff
@@ -154,12 +153,11 @@ def get_conn(self, server_socket):
                 try:
                     os.fstat(fno)
                 except OSError:
-                    # Socket is invalid, close the connection.
+                    # Socket is invalid, close the connection, insert at
+                    # the front.
+                    self.connections.remove(conn)
+                    self.connections.insert(0, conn)
```

Comment:
Inserting things into the very beginning will shift the *entire* array in memory.

Reply:
(https://stackoverflow.com/a/36529893/595220)

Reply:
This isn't an action we would expect to be performing often.